src/ledger/pathfind.ts

```diff
@@ -95,13 +95,25 @@ function filterSourceFundsLowPaths(pathfind: PathFind,
 ): RippledPathsResponse {
   if (pathfind.source.amount &&
       pathfind.destination.amount.value === undefined && paths.alternatives) {
-    paths.alternatives = _.filter(paths.alternatives, alt =>
-        !!alt.source_amount &&
-        !!pathfind.source.amount &&
-        // TODO: Returns false when alt.source_amount is a string. Fix?
-        typeof alt.source_amount !== 'string' &&
-        new BigNumber(alt.source_amount.value).eq(pathfind.source.amount.value)
-    )
+    paths.alternatives = _.filter(paths.alternatives, alt => {
+      let altSourceAmountValue: BigNumber
+      let pathfindSourceAmountValue: BigNumber
+      if (typeof alt.source_amount === 'string') {
+        altSourceAmountValue = new BigNumber(alt.source_amount)
+      } else if (alt.source_amount) {
+        altSourceAmountValue = new BigNumber(alt.source_amount.value)
+      } else {
+        // No alt.source_amount
+        return false
+      }
+      if (pathfind.source.amount.currency === 'XRP') {
+        pathfindSourceAmountValue =
+          new BigNumber(xrpToDrops(pathfind.source.amount.value))
+      } else {
+        pathfindSourceAmountValue = new BigNumber(pathfind.source.amount.value)
+      }
```

Comment:
This could be changed to
```
const pathfindSourceAmountValue = new BigNumber(
  pathfind.source.amount.currency === 'XRP' ?
  xrpToDrops(pathfind.source.amount.value) :
  pathfind.source.amount.value)
```